Gems/ROS2/Code/Source/RobotImporter/Utils/RobotImporterUtils.cpp

```diff
@@ -341,161 +342,224 @@ namespace ROS2::Utils
+
+                // If we successfully matched the prefix, and the replacement path is completely empty, we don't need to look any further.
+                // There's no match.
+                if (replacedUriPath.empty())
                 {
-                    // Strip the number of characters from the Uri scheme from beginning of the path
-                    AZ::IO::PathView strippedUriPath = uriPath.Native().substr(schemePrefix.size());
-                    if (strippedUriPath.empty())
+                    AZ_Trace("ResolveAssetPath", R"(Resolved Path is empty: "%.*s" -> "")" "\n", AZ_PATH_ARG(unresolvedPath));
+                    return "";
+                }
+
+                // If the replaced path is an absolute path, if it exists, return it.
+                // If it doesn't exist, keep trying other replacements.
+                if (replacedUriPath.IsAbsolute())
+                {
+                    if (fileExistsCB(replacedUriPath))
                     {
-                        // The stripped URI path is empty, so there is nothing to resolve
-                        return AZStd::nullopt;
+                        AZ_Trace("ResolveAssetPath", R"(Resolved Absolute Path: "%.*s" -> "%.*s")" "\n", 
+                            AZ_PATH_ARG(unresolvedPath), AZ_PATH_ARG(replacedUriPath));
+                        return replacedUriPath;
                     }
-
-                    // Check to see if the relative part of the URI path refers to a location
-                    // within each <ament prefix path>/share directory
-                    for (const AZ::IO::Path& amentPrefixPath : amentPrefixPaths)
+                    else
                     {
-                        auto pathIter = strippedUriPath.begin();
-                        AZ::IO::PathView packageName = *pathIter;
-                        const AZ::IO::Path amentSharePath = amentPrefixPath / "share";
-                        const AZ::IO::Path packageManifestPath = amentSharePath / packageName / "package.xml";
-
-                        if (const AZ::IO::Path candidateResolvedPath = amentSharePath / strippedUriPath;
-                            fileExistsCB(packageManifestPath) && fileExistsCB(candidateResolvedPath))
-                        {
-                            return candidateResolvedPath;
-                        }
+                        // The file didn't exist, so continue.
+                        continue;
                     }
+                }
 
-                    // The URI path cannot be resolved within the any ament prefix path,
-                    // so try the directory containing the URDF file as well as any of its parent directories
-                    for (const AZ::IO::Path& urdfAncestorPath : urdfAncestorPaths)
+                // The URI path cannot be resolved within the any ament prefix path,
+                // so try the directory containing the root file as well as any of its parent directories
```

Comment:
Can you fix the comment to no longer mention the ament prefix path?
```suggestion
                // The URI path is not absolute, so attempt to append it to the ancestor directories of the URDF/SDF file
```